src/main/java/org/apache/pulsar/io/jcloud/util/AvroRecordUtil.java

```diff
@@ -75,7 +75,7 @@ public static org.apache.pulsar.client.api.Schema<GenericRecord> getPulsarSchema
             Record<GenericRecord> record) {
 
         org.apache.pulsar.client.api.Schema<GenericRecord> schema = record.getSchema();
-        if (record.getValue().getSchemaVersion() == null) {
+        if (record.getValue().getSchemaVersion() == null && record.getValue().getSchemaType() != SchemaType.BYTES) {
```

Comment:
How about adding a test displaying the intended behavior leading to this bug?

Reply:
This is because of the breaking change caused by https://github.com/apache/pulsar/pull/15622

For now, we should use the following way to get schema Info instead of skipping it directly.

```
        if (schema instanceof AutoConsumeSchema) {
            AutoConsumeSchema autoConsumeSchema = (AutoConsumeSchema) schema;
            schema = (Schema<GenericRecord>) autoConsumeSchema.atSchemaVersion(null);
        }

```